File: 0000-0999/0071.SimplifyPath.go
func simplifyPath(path string) string {
    dirs := strings.Split(path, "/")
    i := 0
    for _, dir := range dirs {
        if dir == ".." {
            if i > 0 {
                i--
            }
        } else if dir != "" && dir != "." {
            dirs[i] = dir
            i++
        }
    }
    return "/" + strings.Join(dirs[:i], "/")
}

File: 0000-0999/0071.SimplifyPath.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []
        for p in path.split('/'):
            if p not in ['', '.', '..']:
                stack.append(p)
            elif p == '..' and stack:
                stack.pop()
        return '/' + '/'.join(stack)
        